doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -971,35 +971,43 @@ Generation of induction principles with ``Scheme``
 
 .. cmd:: Scheme {? @ident := } @scheme_kind {* with {? @ident := } @scheme_kind }
 
-   .. insertprodn scheme_kind sort_family
-
    .. prodn::
-      scheme_kind ::= Equality for @reference
-      | {| Induction | Minimality | Elimination | Case } for @reference Sort @sort_family
-      sort_family ::= Set
-      | Prop
-      | SProp
-      | Type
-
-  A high-level tool for automatically generating
-  (possibly mutual) :term:`induction principles <induction principle>`
-  for given types and sorts.
-  Each :n:`@reference` is a different inductive type identifier belonging to
-  the same package of mutual inductive definitions.
-  The command generates the :n:`@ident`\s as mutually recursive
-  definitions. Each term :n:`@ident` proves a general principle of mutual
-  induction for objects in type :n:`@reference`.
-
-  :n:`@ident`
-    The name of the scheme. If not provided, the scheme name will be determined automatically
-    from the sorts involved.
-
-  :n:`Minimality for @reference Sort @sort_family`
-    Defines a non-dependent elimination principle more natural for inductively defined relations.
-
-  :n:`Equality for @reference`
-     Tries to generate a Boolean equality and a proof of the decidability of the usual equality.
-     If :token:`reference` involves other inductive types, their equality has to be defined first.
+      
+      scheme_kind ::= @scheme_type for @reference Sort @sort_family
+      scheme_type ::= {| Induction | Minimality | Elimination | Case }
+      sort_family ::= {| Prop | SProp | Set | Type |}
+
+   Generates (possibly mutual) :term:`induction principles <induction principle>`
+   for given types and sorts. Each :n:`@reference` is a different inductive type
+   identifier belonging to the same package of mutual inductive definitions. The
+   command generates the :n:`@ident`\s as mutually recursive definitions. Each
+   term :n:`@ident` proves a general principle of mutual induction for objects
+   in type :n:`@reference`.
```

Comment:
```suggestion
   Generates :term:`induction principles <induction principle>`
   for the inductive type :n:`@ident` and :n:`@sort_family`. Each :n:`@reference` is a different inductive type
   identifier belonging to the same package of mutual inductive definitions. If
   multiple :n:`@ident`\s are given, the induction principles are be mutually
   recursive Each
   term :n:`@ident` proves a general principle of mutual induction for objects
   in type :n:`@reference`.
```

"term :n:`@ident` proves" - should be "provides"?

By "different", do you mean to generate only one induction principle for each (reference, sort).  How about adding a sentence stating that clearly?  But Coq permits that, e.g.

```
Parameter A B C D : Prop.
Inductive tree : Set :=
| node : A -> forest -> tree
with forest : Set :=
| leaf : B -> forest
| cons : tree -> forest -> forest.
Scheme tree_forest_rec := Induction for tree Sort Set
  with forest_tree_rec := Induction for forest Sort Set.
Scheme tree_forest_rec2 := Induction for tree Sort Set
  with forest_tree_rec2 := Induction for forest Sort Set.
```

Reply:
Right that is how it is intended to be used. I am worried if I don't mention "different" that people might attempt
```coq
Scheme foo := Induction for tree Sort Set
  with bar := Minimality for tree Sort Set.
```
Which isn't allowed but might confuse some users if they don't know that the inductive type has to be different.

Reply:
In this case `with` is only used (and allowed to be used) because tree and forsest are a mutual inductive type (whose syntax also includes "with" which is suggestive). It's not like we are doing multiple scheme commands together but rather one for a mutual inductive type.

Reply:
I've rewritten that paragraph. I was not the original author and the wording was too confusing.